packages/devtools_app/lib/src/inspector/flutter/inspector_screen_details_tab.dart

```diff
@@ -0,0 +1,229 @@
+            Expanded(
+              child: TabBarView(
+                children: tabViews,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class LayoutDetailsTab extends StatefulWidget {
+  const LayoutDetailsTab({Key key, this.controller}) : super(key: key);
+
+  final InspectorTreeControllerFlutter controller;
+
+  @override
+  _LayoutDetailsTabState createState() => _LayoutDetailsTabState();
+}
+
+class _LayoutDetailsTabState extends State<LayoutDetailsTab>
+    with AutomaticKeepAliveClientMixin<LayoutDetailsTab>
+    implements InspectorControllerClient {
+  InspectorTreeControllerFlutter get controller => widget.controller;
+
+  RemoteDiagnosticsNode get selected => controller.selection?.diagnostic;
+
+  // Lifecycle hooks
+  @override
+  void initState() {
+    super.initState();
+    controller.addClient(this);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    controller.removeClient(this);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    if (selected == null)
+      return const Center(
+        child: CircularProgressIndicator(),
+      );
+    if (!selected.isFlex)
+      return Container(
+        // TODO(albertusangga): Visualize non flex widget constraint with Bounding Box model
+        child: const Text('TODOs for Non Flex widget'),
+      );
+    return StoryOfYourFlexWidget(
+      diagnostic: selected,
+      // TODO(albertusangga): Optimize and cache deserialization instead of always calling it when state changes
+      properties: RenderFlexProperties.fromJson(selected.renderObject),
+    );
+  }
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  void onChanged() {
+    setState(() {});
+  }
+
+  @override
+  void scrollToRect(Rect rect) {
+    // do nothing since we are not doing scrolling here
+  }
+}
+
+@immutable
+class StoryOfYourFlexWidget extends StatelessWidget {
+  const StoryOfYourFlexWidget({
+    this.diagnostic,
+    this.properties,
+    Key key,
+  }) : super(key: key);
+
+  final RemoteDiagnosticsNode diagnostic;
+  final RenderFlexProperties properties;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final List<Widget> children = [
+      for (var child in diagnostic.childrenNow)
+        Expanded(
+          child: Container(
+            decoration: BoxDecoration(
+              color: theme.backgroundColor,
+              border: Border.all(
+                color: theme.primaryColor,
+                width: 1.0,
+              ),
+              boxShadow: [
+                BoxShadow(
+                  color: theme.primaryColor,
+                  offset: Offset.zero,
+                  blurRadius: 10.0,
+                )
+              ],
+            ),
+            child: Center(
+              child: Text(child.description),
+            ),
+          ),
+        ),
+    ];
+    final Widget flexWidget = properties.type == Row
```

Comment:
instead of
```dart
   final Widget flexWidget = properties.type == Row
        ? Row(children: children)
        : Column(children: children);
```
you can write
```dart
final Flex(direction: properties.direction, children: children);
```

Reply:
also, instead of `flexWidget`, maybe name this `widgetVisualization` or some name that indicates what this widget represents.

Reply:
absolutely, this makes it more elegant